### ISSUE TITLE
scripts do not take arguments as in the REPL

### ISSUE THREAD

**Issue by amtoine** [nushell/nushell#9766]
### Describe the bug

when running a script, it appears quotes, double quotes and backticks are removed from arguments before being passed down to `main`.

### How to reproduce

1. put the following in your `$env.PATH` as an executable file called `foo`
```nushell
#!/usr/bin/env nu

export def main [...args: string] {
    print $args
}
```
2. run the following command
```nushell
foo "-a" 123 '-b' 456 `-c` 789
```
3. see the following error
```
Error: nu::parser::unknown_flag

  × The `main` command doesn't have flag `-a`.
   ╭─[<commandline>:1:1]
 1 │ main -a 123 -b 456 -c 789
   ·       ┬
   ·       ╰── unknown flag
   ╰────
  help: Available flags: --help(-h). Use `--help` for more information.
```

### Expected behavior

i expected it to work as in the REPL, i.e.
```nushell
def foo [...args: string] {
    print $args
}
```
and
```nushell
> foo "-a" 123 '-b' 456 `-c` 789
╭───┬─────╮
│ 0 │ -a  │
│ 1 │ 123 │
│ 2 │ -b  │
│ 3 │ 456 │
│ 4 │ -c  │
│ 5 │ 789 │
╰───┴─────╯
```

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.82.1                                   |
| branch             | main                                     |
| commit_hash        | 656f707a0be48d04c356cc307450213ef1a39619 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16)      |
| rust_channel       | 1.69.0-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12)      |
| build_time         | 2023-07-17 19:32:26 +02:00               |
| build_rust_channel | release                                  |
| allocator          | standard                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  | hist, plot                               |

### Additional context

i'm trying to run an external command with arguments from a script.

hope i'm not completely forgetting about how to pass arguments :eyes:

**Comment by amtoine** [comment]
it appears writting
```nushell
#!/usr/bin/env nu

extern-wrapped main [...args: string] {
    print $args
}
```
solves the issue :open_mouth:

**Comment by mengsuenyan** [comment]
the main function just like the normal custom command,

file: foo.nu
```nu
export def main [
    --a (-a): number,
    --b (-b): number,
    --c (-c): number,
] {
    $a + $b + $c
}
```

```nu
> use foo.nu
> foo -a 1 -b 2 -c 3
```

**Comment by amtoine** [comment]
@mengsuenyan 
yes, your example works fine but i do not think it solves this issue :confused: 

we can see that
```nushell
> def foo [...args] { print $args }; foo "-a"
╭───┬────╮
│ 0 │ -a │
╰───┴────╯
```
works fine in the REPL but the exact same command in a script fails.
it looks like the quotes are removed when passing the arguments to the script, but `"-a"` is a string, not an invalid option `-a`.
